src/main/java/com/lineate/testyourlexicon/controllers/UserController.java

```diff
@@ -0,0 +1,26 @@
+package com.lineate.testyourlexicon.controllers;
+
+import com.lineate.testyourlexicon.dto.UserRegistrationDto;
+import com.lineate.testyourlexicon.services.UserService;
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import net.sf.saxon.type.ValidationException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+@AllArgsConstructor
+public class UserController {
+
+  private final UserService userService;
+
+  @PostMapping("/register")
```

Comment:
According to REST principles this endpoint should be just POST /users. You can read more about REST endpoints naming here: https://restfulapi.net/resource-naming/